src/sqlfluff/core/parser/helpers.py

```diff
@@ -28,7 +28,7 @@ def check_still_complete(
     initial_str = join_segments_raw(segments_in)
     current_str = join_segments_raw(matched_segments + unmatched_segments)
 
-    if initial_str != current_str:
+    if initial_str != current_str:  # pragma: nocover
```

Comment:
Was this code previously covered by a test? Is it possible this `if` could be removed?

Reply:
@barrywhart so it was originally being hit by the test I removed:
![image](https://user-images.githubusercontent.com/80432516/146442443-fc188f06-da8c-4526-ac34-d0911c89ba5a.png)
which shouldn't have been a test in the first place.

It's one of those awkward cases where this isn't getting touched anywhere else other than this random test case, and I can't immediately think of a situation that would trigger it, but think it's probably safest to to leave it in there as it may catch some other parsing issues that aren't in the tests?

Reply:
That makes sense -- I was just curious. 👍🏽

Reply:
@barrywhart no worries, took the pragma(tic) approach 👍 Want to do the honours merging? 😛